### DISCUSSION TITLE
OAuth2 "password flow" deprecated, is there other way to implement login process(not in FastAPI tutorials "Securities" because of "deprecated" examples)

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#6225] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
no code here
```

### Description

FastAPI tutorial series are very useful and helpful for me (as beginner and newcommer).
But when consulting "Securities" parts, the way login process are implemented is "OAuth2 Password flow(grant)"
I knew that OAuth2 Password flow is "deprecated" because of several less secure reasons after visiting OAuth2 official website.

Is there still encouraged(or okay) to practice and implement the securities parts as the FastAPI tutorials says?
or is there more secure and better solutions to that? if it is, is there a plan to update the tutorials or let me know where to consult...

Anyway, really appreaciate for making Python backend more joyful and  take-it-to-next-level !

### Operating System

Windows

### FastAPI Version

latest

### Python Version

3.10

**Comment by JarroVGIT** [comment, marked as answer]
As of now, I would absolutely recommend to follow the tutorials. They give you several good ideas on how to implement FastAPI concepts (such as `Depends`, among others) and give a general idea about Oauth2 implementation considerations (such as, JWT, password salting, etc). I had a blast going through them and reading up on the details on different aspects. For example, I have a better grasp of Oauth2, the actors involved and their roles (and terminology used) because I followed and tried to fully comprehend the FastAPI Advanced Security tutorial. 

So, will it give you a production ready login system? No, it won't. But you can get the bits and pieces from it that can help you get there!

Also, there is not exactly something wrong with the password flow if you just use it for your own website. However, if you want third parties to be able to log in users using your user-system, you should definitely avoid it. You would never want a user to enter credentials to your system in someone else's front end (as you don't know what that front end will do with those credentials). 

So, in short; definitely go through the tutorial and Google every concept you don't fully understand, and the tutorial will be a fantastic guide in your learning path :) 

Edit: I just wanted to make sure to emphasize that in order to build a login system with Fastapi, you do not need Oauth2 per se (or actually, at all). But, since the password flow in oauth2 basically gives a basic login system, you can still use that and not use Oauth for the rest of your application or login system.

**Comment by ghost** [comment]
Really appreicate for your invaluable comments! I totally understand what you point
